x-pack/plugins/infra/server/services/metrics_explorer_views/metrics_explorer_views_client.test.ts

```diff
@@ -0,0 +1,262 @@
+        createMetricsExplorerViewMock('0', {
+          isDefault: true,
+        } as MetricsExplorerViewAttributes),
+      ];
+
+      infraSources.getSourceConfiguration.mockResolvedValue(basicTestSourceConfiguration);
+
+      savedObjectsClient.find.mockResolvedValue({
+        total: 2,
+        saved_objects: [],
+        per_page: 1000,
+        page: 1,
+      });
+
+      const metricsExplorerViewList = await metricsExplorerViewsClient.find({});
+
+      expect(savedObjectsClient.find).toHaveBeenCalled();
+      expect(metricsExplorerViewList).toEqual(metricsExplorerViewListMock);
+    });
+  });
+
+  it('.get resolves the an metrics explorer view by id', async () => {
+    const { metricsExplorerViewsClient, infraSources, savedObjectsClient } =
+      createMetricsExplorerViewsClient();
+
+    const metricsExplorerViewMock = createMetricsExplorerViewMock('custom_id', {
+      name: 'Custom',
+      isDefault: false,
+      isStatic: false,
+    } as MetricsExplorerViewAttributes);
+
+    infraSources.getSourceConfiguration.mockResolvedValue(basicTestSourceConfiguration);
+
+    savedObjectsClient.get.mockResolvedValue({
+      ...metricsExplorerViewMock,
+      type: metricsExplorerViewSavedObjectName,
+      references: [],
+    });
+
+    const metricsExplorerView = await metricsExplorerViewsClient.get('custom_id', {});
+
+    expect(savedObjectsClient.get).toHaveBeenCalled();
+    expect(metricsExplorerView).toEqual(metricsExplorerViewMock);
+  });
+
+  describe('.create', () => {
+    it('generate a new metrics explorer view', async () => {
+      const { metricsExplorerViewsClient, savedObjectsClient } = createMetricsExplorerViewsClient();
+
+      const metricsExplorerViewMock = createMetricsExplorerViewMock('new_id', {
+        name: 'New view',
+        isStatic: false,
+      } as MetricsExplorerViewAttributes);
+
+      mockFindMetricsExplorerList(savedObjectsClient);
+
+      savedObjectsClient.create.mockResolvedValue({
+        ...metricsExplorerViewMock,
+        type: metricsExplorerViewSavedObjectName,
+        references: [],
+      });
+
+      const metricsExplorerView = await metricsExplorerViewsClient.create({
+        name: 'New view',
+      } as CreateMetricsExplorerViewAttributesRequestPayload);
+
+      expect(savedObjectsClient.create).toHaveBeenCalled();
+      expect(metricsExplorerView).toEqual(metricsExplorerViewMock);
+    });
+
+    it('throws an error when a conflicting name is given', async () => {
+      const { metricsExplorerViewsClient, savedObjectsClient } = createMetricsExplorerViewsClient();
+
+      mockFindMetricsExplorerList(savedObjectsClient);
+
+      await expect(
+        async () =>
+          await metricsExplorerViewsClient.create({
+            name: 'Custom',
+          } as CreateMetricsExplorerViewAttributesRequestPayload)
+      ).rejects.toThrow('A view with that name already exists.');
+    });
+  });
+
+  describe('.update', () => {
```

Comment:
These tests don't validate that piece of logic because the gate for the accepted values is at the endpoint level, when the parameters are validated, it shouldn't even reach the client if the value is not acceptable.